src/II/Implementations/LinqRepository.cs

```diff
@@ -34,88 +34,241 @@ protected LinqRepository(ISessionProvider sessionProvider)
-            => Execute(nameof(FindPaged), () => FindPagedInternal(pageIndex, pageSize, func));
+        /// <inheritdoc cref="Repository{TRoot,TId}.FindByIdsInternal" />
+        [NotNull]
+        [ItemNotNull]
+        protected override IEnumerable<TRoot> FindByIdsInternal([NotNull] IEnumerable<TId> ids)
+            => FindInternal(() => CreateQueryable().Where(e => ids.Contains(e.Id)), null, null);
 
+        /// <inheritdoc cref="Get{TResult}" />
         [CanBeNull]
-        protected virtual TRoot GetInternal([NotNull] Func<IQueryable<TRoot>, IQueryable<TRoot>> func)
+        protected virtual TResult GetInternal<TResult>(
+            [NotNull] Func<IQueryable<TRoot>, IQueryable<TResult>> lambda)
         {
             try
             {
-                return func.Invoke(CreateQueryable()).SingleOrDefault();
+                return lambda.Invoke(CreateQueryable()).SingleOrDefault();
             }
             catch (EmptyResultException)
             {
-                return null;
+                return default;
             }
         }
 
+        /// <inheritdoc cref="GetAtIndex{TResult}" />
         [CanBeNull]
-        protected virtual TRoot GetAtIndexInternal([NotNull] Func<IQueryable<TRoot>, IQueryable<TRoot>> func, int index)
+        protected virtual TResult GetAtIndexInternal<TResult>(
+            [NotNull] Func<IQueryable<TRoot>, IQueryable<TResult>> lambda,
+            int index)
         {
             try
             {
-                return func.Invoke(CreateQueryable()).Skip(index).Take(1).SingleOrDefault();
+                return lambda.Invoke(CreateQueryable()).Skip(index).Take(1).SingleOrDefault();
             }
             catch (EmptyResultException)
             {
-                return null;
+                return default;
             }
         }
 
-        protected override IList<TRoot> FindAllInternal()
-            => FindInternal(null);
-
-        [NotNull]
-        protected virtual IList<TRoot> FindInternal([CanBeNull] Func<IQueryable<TRoot>, IQueryable<TRoot>> func)
-            => FindInternal(func, null, null);
-
-        [NotNull]
-        protected virtual IList<TSubType> FindInternal<TSubType>([NotNull] Func<IQueryable<TRoot>, IQueryable<TSubType>> func)
-            => FindInternal(func, null, null);
-
-        [NotNull]
-        protected virtual IList<TRoot> FindInternal([CanBeNull] Func<IQueryable<TRoot>, IQueryable<TRoot>> func, int? skip, int? count)
-            => FindInternal(() => func != null ? func(CreateQueryable()) : CreateQueryable(), skip, count);
-
-        [NotNull]
-        protected virtual IList<TSubType> FindInternal<TSubType>([NotNull] Func<IQueryable<TRoot>, IQueryable<TSubType>> func, int? skip, int? count)
-            => FindInternal(() => func(CreateQueryable()), skip, count);
+        // TODO, i don't get it why <inheritdoc cref="Find{TResult}(Func{IQuerable{TRoot},IQuerable{TResult}},int?,int?)" />doesn't work
```

Comment:
Probably not working because the type of the parameters from the referenced function are different.

Func{IQuerable{TRoot},IQuerable{TResult}} vs Func{IQuerable{TResult}}